carbon/hal/atmegau2.c

```diff
@@ -8,17 +8,17 @@ int carbon_select_u2_gpio(struct _lf_device *device) {
 	struct _carbon_context *context = device->_ctx;
 	lf_assert(context, failure, E_NULL, "No context for selected carbon device.");
 	struct _lf_device *u2 = context->_u2;
-	LF_MODULE_SET_DEVICE_AND_ID(_gpio, u2, _gpio_id);
+	_gpio.index = _gpio_id;
 	return lf_success;
 failure:
 	return lf_error;
 }
 
 int carbon_select_atmegau2(struct _lf_device *device) {
-	LF_MODULE_SET_DEVICE_AND_ID(_button, device, _button_id);
-//	LF_MODULE_SET_DEVICE_AND_ID(_gpio, device, _gpio_id);
-	LF_MODULE_SET_DEVICE_AND_ID(_led, device, _led_id);
-	LF_MODULE_SET_DEVICE_AND_ID(_uart0, device, _uart0_id);
-	LF_MODULE_SET_DEVICE_AND_ID(_wdt, device, _wdt_id);
+	_button.index = _button_id;;
+//	_gpio.index = _gpio_id;;
+	_led.index = _led_id;;
+	_uart0.index = _uart0_id;;
+	_wdt.index = _wdt_id;;
```

Comment:
Is there a reason you have two semicolons on all of these lines?

Reply:
Rip. Fixing.